edx_sysadmin/views.py

```diff
@@ -3,9 +3,108 @@
 """
 
 from django.views.generic.base import TemplateView
+from django.contrib.auth.decorators import login_required
+from django.http import Http404, HttpResponse
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.utils.html import escape
+from django.utils.translation import ugettext as _
+from django.views.decorators.cache import cache_control
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import condition
+from django.views.generic.base import TemplateView
+from opaque_keys.edx.keys import CourseKey
+from django.shortcuts import render
+from .utils.markup import HTML
+from six import StringIO, text_type
+from .utils.utility import get_course_by_id
+
+from xmodule.modulestore.django import modulestore
 
 
-class IndexPage(TemplateView):
-    """View to show the Index page of SysAdmin."""
+class SysadminDashboardView(TemplateView):
+    """View to show the Dashboard page of SysAdmin."""
 
     template_name = "edx_sysadmin/base.html"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize base sysadmin dashboard class with modulestore,
+        modulestore_type and return msg
+        """
+
+        self.def_ms = modulestore()
+        self.msg = u''
+        self.datatable = []
+        super(SysadminDashboardView, self).__init__(**kwargs)
+
+    @method_decorator(ensure_csrf_cookie)
+    @method_decorator(login_required)
+    @method_decorator(cache_control(no_cache=True, no_store=True,
+                                    must_revalidate=True))
+    @method_decorator(condition(etag_func=None))
+    def dispatch(self, *args, **kwargs):
+        return super(SysadminDashboardView, self).dispatch(*args, **kwargs)
+ 
+    def get_courses(self):
+        """ Get an iterable list of courses."""
+
+        return self.def_ms.get_courses()
+
+
+class Courses(SysadminDashboardView):
+    """
+    This manages deleting courses.
+    """
+
+    template_name = "edx_sysadmin/courses.html"
+
+    def get(self, request):
+        """Displays forms and course information"""
+
+        if not request.user.is_staff:
+            raise Http404
+
+        context = {}
+        return render(request, self.template_name, context)
+
+    def post(self, request):
+        """Handle delete action from courses view"""
+
+        if not request.user.is_staff:
+            raise Http404
+
+        action = request.POST.get('action', '')
+        courses = {course.id: course for course in self.get_courses()}
+        if action == 'del_course':
+            course_id = request.POST.get('course_id', '').strip()
+            course_key = CourseKey.from_string(course_id)
+            course_found = False
+            if course_key in courses:
+                course_found = True
+                course = courses[course_key]
+            else:
```

Comment:
Can we skip all of this processing and just call the `get_course_by_id(course_key)` function directly?